include/NovelRT.Interop/Animation/NrtSpriteAnimatorFrame.h

```diff
@@ -1,27 +1,28 @@
-// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
 
 #ifndef NOVELRT_INTEROP_ANIMATION_SPRITEANIMATORFRAME_H
 #define NOVELRT_INTEROP_ANIMATION_SPRITEANIMATORFRAME_H
 
-
+#include "../Graphics/NrtGraphicsTypedefs.h"
 #include "../NrtInteropUtils.h"
 #include "../Timing/NrtTimestamp.h"
-#include "../Graphics/NrtGraphicsTypedefs.h"
 #include "NrtAnimationTypedefs.h"
 
 #ifdef __cplusplus
-extern "C" {
+extern "C"
+{
 #endif
 
-NrtSpriteAnimatorFrame Nrt_SpriteAnimatorFrame_create();
+    NrtSpriteAnimatorFrame Nrt_SpriteAnimatorFrame_create();
 
-NrtResult Nrt_SpriteAnimatorFrame_getTexture(NrtSpriteAnimatorFrame frame, NrtTexture* outputTexture);
-NrtResult Nrt_SpriteAnimatorFrame_setTexture(NrtSpriteAnimatorFrame frame, NrtTexture texture);
-NrtTimestamp Nrt_SpriteAnimatorFrame_getDuration(NrtSpriteAnimatorFrame frame);
-NrtResult Nrt_SpriteAnimatorFrame_setDuration(NrtSpriteAnimatorFrame frame, NrtTimestamp timestamp);
+    NrtResult Nrt_SpriteAnimatorFrame_getTexture(NrtSpriteAnimatorFrame frame, NrtTexture* outputTexture);
+    NrtResult Nrt_SpriteAnimatorFrame_setTexture(NrtSpriteAnimatorFrame frame, NrtTexture texture);
+    NrtTimestamp Nrt_SpriteAnimatorFrame_getDuration(NrtSpriteAnimatorFrame frame);
+    NrtResult Nrt_SpriteAnimatorFrame_setDuration(NrtSpriteAnimatorFrame frame, NrtTimestamp timestamp);
 
-NrtResult Nrt_SpriteAnimatorFrame_addFrameEnter(NrtSpriteAnimatorFrame frame, void(*func)());
-NrtResult Nrt_SpriteAnimatorFrame_addFrameExit(NrtSpriteAnimatorFrame frame, void(*func)());
+    NrtResult Nrt_SpriteAnimatorFrame_addFrameEnter(NrtSpriteAnimatorFrame frame, void (*func)());
```

Comment:
Do we want to keep these sorts of spaces in fnptrs?

Reply:
Based on my testing, `PointerAlignment` is already set to `Left` and clang-format is not picking this up as a C-Style cast (since it really isn't...) so this will be unavoidable unfortunately.

At the moment, I'm using clang-format from Clang 10 since that is what our CI is targeting iirc. If that changes, I can certainly check the newer style guides as it updates to see if they have anything to address this in the future.

Reply:
I think we should keep it and we should prefer typedefs rather than inline fnptr declarations.

Reply:
Function pointers are pretty hard to understand as is, so I totally approve of extracting them into typedefs.

Reply:
In before:
```cpp
template<typename... TArgs>
typedef void (*Action)(TArgs... args);

template<typename... TArgs, typename TResult>
typedef TResult (*Func)(TArgs... args);
```

Reply:
At that point, just use std::function :stuck_out_tongue:

Reply:
TODO for me: Update this to typedefs (outside of this PR)